ingest-asset-reconciler/src/main/scala/uk/gov/nationalarchives/Lambda.scala

```diff
@@ -39,34 +42,107 @@ class Lambda extends LambdaRunner[Input, StateOutput, Config, Dependencies] {
 
   private def stripFileExtension(title: String) = title.split('.').dropRight(1).mkString(".")
 
+  private def coTitleMatchesAssetChildTitle(potentialCoTitle: Option[String], assetChild: FileDynamoTable): Boolean =
+    potentialCoTitle.exists { titleOfCo => // DDB titles don't have file extensions, CO titles do
+      lazy val fileNameWithoutExtension = assetChild.name
+      val potentialAssetChildTitleOrFileName = assetChild.title.getOrElse("")
+      val assetChildTitleOrFileName = if potentialAssetChildTitleOrFileName.isEmpty then fileNameWithoutExtension else potentialAssetChildTitleOrFileName
+
+      val numOfDotsInTitleOrFileName = assetChildTitleOrFileName.count(_ == '.')
+      val numOfDotsInTitleOfCo = titleOfCo.count(_ == '.')
+      val differenceInNumberOfDots = numOfDotsInTitleOrFileName - numOfDotsInTitleOfCo
+
+      val (titleOfCoWithoutExtension, assetChildTitleOrFileNameWithoutExtension) =
+        if (numOfDotsInTitleOrFileName == numOfDotsInTitleOfCo) (titleOfCo, assetChildTitleOrFileName)
+        else if (differenceInNumberOfDots == 1) (titleOfCo, stripFileExtension(assetChildTitleOrFileName))
+        else if (abs(differenceInNumberOfDots) > 1) (titleOfCo, assetChildTitleOrFileName) // then let it fail the equality comparison below
+        else (stripFileExtension(titleOfCo), assetChildTitleOrFileName)
+
+      titleOfCoWithoutExtension == assetChildTitleOrFileNameWithoutExtension
+    }
+
+  private def verifyFilesInDdbAreInPreservica(
+      childrenForRepresentationType: List[FileDynamoTable],
+      bitstreamInfoPerContentObject: Seq[BitStreamInfo],
+      assetId: UUID,
+      representationType: RepresentationType
+  ) = {
+    val childrenThatDidNotMatchOnChecksum =
+      childrenForRepresentationType.filter { assetChild =>
+        val bitstreamWithSameChecksum = bitstreamInfoPerContentObject.find { bitstreamInfoForCo =>
+          assetChild.checksumSha256 == bitstreamInfoForCo.fixity.value &&
+          coTitleMatchesAssetChildTitle(bitstreamInfoForCo.potentialCoTitle, assetChild)
+        }
+
+        bitstreamWithSameChecksum.isEmpty
+      }
+
+    lazy val idsOfChildrenThatDidNotMatchOnChecksum = childrenThatDidNotMatchOnChecksum.map(_.id)
```

Comment:
Howcome this is `lazy val`? Also, do you need this interim variable at all, it's only used once in the output and you could put `childrenThatDidNotMatchOnChecksum.map(_.id)` straight in the string. 
The variable name is very long and it might tidy it up a bit, what do you think?

Reply:
With this code, I moved what was already there, to methods but also decided to do a minor clean up

1 . This line was initially in the `else` block, so I moved it here so that I could remove the braces from the `else` to make it cleaner; it's a `lazy val` so that it's not run in the case that `childrenThatDidNotMatchOnChecksum.isEmpty` i.e. it only runs the code when used
2. I'm not sure why I saved it to a variable, maybe it was used more than once initially but yes, it's a better idea to add it directly to the string and then we no longer have the long variable name